generate.py

```diff
@@ -106,6 +106,30 @@ def generate_message(media, target, conf, result):
         raise RuntimeError('Invalid media {} for {}'.format(media, target))
 
 
+async def handle_target(resolvers, name, target):
+    result = await resolve_target(target, resolvers)
+    msg = "none"
+
+    if any(
+            success
+            for host, rs in result.items()
+            for resolver, servers in rs.items()
+            for server, success in servers.items()
+    ):
+        msg = "some"
+
+    if all(
+            success
+            for host, rs in result.items()
+            for resolver, servers in rs.items()
+            for server, success in servers.items()
```

Comment:
duplication, why not save the result temporarily and reorder to if all(tmp) else if any(tmp)?